src/test/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSparkUnitTest.java

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceWindowFunctions;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Unit tests for FindBadGenomicKmersSpark
+ */
+public class FindBadGenomicKmersSparkUnitTest extends BaseTest {
+
+    private static final int KMER_SIZE = SVConstants.KMER_SIZE;
+    private static final String REFERENCE_FILE_NAME = hg19MiniReference;
+
+    @Test(groups = "spark")
+    public void badKmersTest() throws IOException {
+        final File refSequenceFile = File.createTempFile("genomic_sequence", "txt");
+        refSequenceFile.deleteOnExit();
+
+        final String polyA = StringUtils.repeat('A', KMER_SIZE);
+        final String polyC = StringUtils.repeat('C', KMER_SIZE);
+        final String polyT = StringUtils.repeat('T', KMER_SIZE);
+
+        try ( PrintWriter writer = new PrintWriter(refSequenceFile) ) {
+            long nTimes = FindBadGenomicKmersSpark.MAX_KMER_FREQ;
+            // write polyA and polyC the max number of times possible while evading the trigger
+            while ( nTimes-- > 0L ) {
+                writer.println(polyA);
+                writer.println(polyC);
+            }
+            // tip polyA over the edge by writing its reverse complement
+            writer.println(polyT);
+        }
+
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        final List<SVKmer> badKmers = FindBadGenomicKmersSpark.processRefFile(ctx, refSequenceFile);
+
+        // should have just one bad kmer:  polyA
+        Assert.assertEquals(badKmers.size(), 1);
+        Assert.assertEquals(badKmers.get(0).toString(KMER_SIZE), polyA);
+
+        if ( !refSequenceFile.delete() )
```

Comment:
I appreciate the aim of non-interference.  I think it's unnecessary though.  We enforce isolation through unique output file names.  If that is insufficient, deleting them immediately probably isn't sufficient either since there's a good chance that aberrant shutdown could leave them in place, or tests running in parallel could break each other.